File: isma-next-hsm/src/main/kotlin/ru/nstu/isma/hsm/common/IndexProvider.java
package ru.nstu.isma.hsm.common;

/**
 * @author Maria Nasyrova
 * @since 06.10.2015
 */
public interface IndexProvider {

    String getDifferentialArrayCode(String code);

    String getAlgebraicArrayCode(String code);

    String getAlgebraicArrayCodeForDifferentialEquation(String aeCode);

}

File: isma-next-hsm/src/main/kotlin/ru/nstu/isma/hsm/common/IndexProvider.kt
package ru.nstu.isma.hsm.common

/**
 * @author Maria Nasyrova
 * @since 06.10.2015
 */
interface IndexProvider {
    fun getDifferentialArrayCode(code: String?): String
    fun getAlgebraicArrayCode(code: String?): String
    fun getAlgebraicArrayCodeForDifferentialEquation(aeCode: String?): String
}